### PR TITLE
Use appropriate CTA

### PR DESCRIPTION
**Pull request by kevin-chassagne**
useMetamaskDetect hook

### CODE REVIEW

**Comment by kevin-chassagne** [comment]
@tannn-younet, @cuonglt-ync can you review it ?

**Comment by nkkurt** [comment]
@tannn-younet @viphan007 Could you review this today? Thank you!

**Review by nkkurt — APPROVED** [review]
LGTM, but please wait for a dev review before merging.

**Comment by jlazoff** [comment]
Hi @cuonglt-ync @tannn-younet @viphan007 ,

Can you please review,
Approve and merge this PR. Your code review is appreciated. 

Thank you 🙏🏻

File: src/components/HeroContainer.js
Comment (viphan007) [inline src/components/HeroContainer.js:104]: I think putting ref here may not be necessary

File: src/components/FullWidthCta.js
Comment (viphan007) [inline src/components/FullWidthCta.js:136]: This module support multiple ctas in many places (Ex: The institutions page have this module with 3 ctas and does not depend on MetaMask extension)
So this code will break that, we need an extra condition

File: src/components/HeroContainer.js
Comment (kevin-chassagne) [reply src/components/HeroContainer.js:104]: It avoid the es-lint warning

File: src/components/FullWidthCta.js
Comment (kevin-chassagne) [reply src/components/FullWidthCta.js:136]: Thank you for noticing this! I'll fix that

**Comment by viphan007** [comment]
Hi @kevin-chassagne,
I have tested and got some issues to consider (with MM extension installed)

1. A full page load (without client-side navigation) on other pages (not home page), description and images were lost in hero module
Screenshot 2023-12-13 at 17 43 01

2. Home page shows wrong content when navigating client-side from another page until it is reloaded: https://drive.google.com/file/d/1xLV1se7YvqA_en8EdqnsQD0jxn8k5P_v/view?usp=sharing

**Comment by kevin-chassagne** [comment]
> Hi @kevin-chassagne, I have tested and got some issues to consider (with MM extension installed)
> 
> 1. A full page load (without client-side navigation) on other pages (not home page), description and images were lost in hero module
> 
> Screenshot 2023-12-13 at 17 43 01
> 2. Home page shows wrong content when navigating client-side from another page until it is reloaded: https://drive.google.com/file/d/1xLV1se7YvqA_en8EdqnsQD0jxn8k5P_v/view?usp=sharing

Fixed, I still have to fix the build with the new context provider i've created

File: src/components/FullWidthCta.js
Comment (kevin-chassagne) [reply src/components/FullWidthCta.js]: Hey @viphan007 
Can you review the code again after my updates ? 
Sorry for asking you this twice

**Comment by kevin-chassagne** [comment]
@viphan007 Everything should be fixed now, sorry for this!

**Comment by nkkurt** [comment]
Thank you so much for the review and catching these @viphan007 ! 🙏

**Comment by nkkurt** [comment]
@kevin-chassagne Could you keep the Download CTA in
- https://metamask.consensys.io/swaps/ We need to update the copy before we can update the CTA here. Let's leave the original CTA for now. The CTA on this page will be updated to Swap, as part of another ticket. 
Screenshot 2023-12-13 at 2 48 37 PM

**Comment by nkkurt** [comment]
And on https://metamask.consensys.io/buy-crypto/

Let's revert the CTA update on this page. We need to update the Hero CTA, copy and visual as well while updating CTAs on this page and we need support form design and copy teams before we can do that.

Screenshot 2023-12-13 at 2 57 19 PM

Screenshot 2023-12-13 at 3 00 22 PM

**Comment by nkkurt** [comment]
Let's also not update the CTA on the NFT page. There's no NFT functionality in portfolio. We may want to update the CTA to something else here. We'll handle it as a separate ticket.

Screenshot 2023-12-13 at 3 29 40 PM

File: src/components/FullWidthCta.js
Comment (viphan007) [reply src/components/FullWidthCta.js]: Hi @kevin-chassagne, it worked properly

**Comment by viphan007** [comment]
Hi @kevin-chassagne,

I made a commit to fix the hero description and rename the fragment ContentfulCtaBaseFields.

The only thing that concern me is that the hero content is flashing 2 times whether the metamask extension is installed or not (production mode). I tried to solve this issue but the useMetamaskDetect using useEffect confused me

https://drive.google.com/file/d/1Roa601MzGh-FsBpKd6_IzTGvKLYTOcJz/view?usp=sharing

**Comment by kevin-chassagne** [comment]
Thanks @viphan007! 
I tried this in staging: 
```  const { childMarkdownRemark: { html: htmlPortfolio } = {} } =
    descriptionPortfolio || {}``` 
I don't know why it didn't work. Maybe the time for the backend to populate the data🤷‍♂️

I tried to got no flashing too but we have to set isMetaMaskDetected to `false` first (for Safari and FF) and after the event is dispatch we can know if there is metamask installed... And a setTimeout is not really clean... So I have to think about a better solution to avoid flash

**Comment by nkkurt** [comment]
Did we change anything on production that could change the existing behavior?
Screenshot 2023-12-14 at 4 36 44 PM

**Comment by nkkurt** [comment]
Yes, we need to fix the flash indeed.

**Comment by kevin-chassagne** [comment]
> Did we change anything on production that could change the existing behavior?
We are speaking of the button in the header ? The only changes we have done are in this branch and the code is not merged yet... It could be from Contentful.

**Comment by kevin-chassagne** [comment]
Hey @viphan007 
Can you review the updated code please ? I hope I've fixed everything 🤞

File: src/components/layout.scss
Comment (viphan007) [inline src/components/layout.scss:1209]: Hi @kevin-chassagne,
opacity:0 on body seam has issue with lighthouse, i tried with 0.01 and it's ok. I hope that it doesn't affect SEO performance

File: src/hooks/useMetamaskDetect.js
Comment (viphan007) [inline src/hooks/useMetamaskDetect.js:10]: This slows down the initial page rendering, can we reduce it?

Otherwise I think everything is good, i think we can merge and improve later

Comment (kevin-chassagne) [reply src/hooks/useMetamaskDetect.js:10]: I'm going to test and see how much I can reduce it to.

Comment (kevin-chassagne) [reply src/hooks/useMetamaskDetect.js]: Reduced to 50ms, it's seems to be enough 👌

**Review by viphan007 — APPROVED** [review]

**Comment by viphan007** [comment]
Thanks @kevin-chassagne, everything looks good

**Pull request merged**
